subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -315,6 +315,9 @@ void lwm2m_reset_message(struct lwm2m_message *msg, bool release)
 	} else {
 		msg->message_timeout_cb = NULL;
 		(void)memset(&msg->cpkt, 0, sizeof(msg->cpkt));
+#if defined(CONFIG_LWM2M_RESOURCE_DATA_CACHE_SUPPORT)
```

Comment:
It just struck me, we should add `CONFIG_LWM2M_RESOURCE_DATA_CACHE_SUPPORT` to `tests/net/all/prj.conf`, otherwise this variant won't be built by the CI (easy to introduce regressions).

Reply:
OK. How we manage dependencies
_depends on (LWM2M_RW_SENML_JSON_SUPPORT || LWM2M_RW_SENML_CBOR_SUPPORT)
depends on (POSIX_CLOCK && RING_BUFFER):_

Reply:
Or was a idea set CONFIG_LWM2M_RESOURCE_DATA_CACHE_SUPPORT=n?

Reply:
I added CONFIG_LWM2M_RESOURCE_DATA_CACHE_SUPPORT=y at PR it not triggered any issue. @rlubos Is that correct implementation about what did you request?

Reply:
yes, that's what I had in mind.